sources/packages/backend/libs/integrations/src/cas/cas.service.ts

```diff
@@ -129,26 +131,62 @@ export class CASService {
     supplierData: CreateSupplierAndSiteData,
   ): Promise<CreateSupplierAndSiteResponse> {
     const url = `${this.casIntegrationConfig.baseUrl}/cfs/supplier/`;
+    return this.processSupplierAndSite(url, supplierData);
+  }
+
+  /**
+   * Create supplier site for existing supplier.
+   * @param supplierData data to be used for supplier and site creation.
+   * @returns submitted data and CAS response.
+   */
+  async createExistingSupplierAndSite(
+    supplierData: CreateExistingSupplierAndSiteData,
+  ): Promise<CreateSupplierAndSiteResponse> {
+    const url = `${this.casIntegrationConfig.baseUrl}/cfs/supplier/${supplierData.supplierNumber}/site`;
+    return this.processSupplierAndSite(url, supplierData);
+  }
+
+  /**
+   * Process the API call to create supplier and site.
+   * @param url API url to create a new or existing supplier and site.
+   * @param submittedData data submitted to the API.
+   * @returns submitted data and CAS response
+   */
+  async processSupplierAndSite(
```

Comment:
I would definitely be against a single method doing the following tasks for 2 different API calls.

1. Building the Payload
2. Reading and parsing the response.
3. Error handling

as these things can be potentially different for different APIs and needs to delegated to an individual method. 

On the other side, this is what I would recommend. 

New method call create site API.

```ts
 async createSiteForExistingSupplier(
    supplierData: CreateExistingSupplierSiteData,
  ): Promise<CreateExistingSupplierSiteResponse> 
```

Common ptivate methpod to build address.